hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java

```diff
@@ -347,6 +348,15 @@ public boolean isLocked(K key) {
         return result;
     }
 
+    @Override
+    public void setTTL(K key, long ttl, TimeUnit timeunit) {
+        if (getNodeEngine().getClusterService().getClusterVersion().isLessThan(Versions.V3_11)) {
```

Comment:
If we add this version check inside `default-record-store`, client initiated operations can also be caught.